### PR TITLE
Add test for stdlib coverage.

### PR DESCRIPTION
**Pull request by hugopl**
Use -u to update the list of files expected to fail.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,19 +2,32 @@ name: CI
...
-        run: .github/workflows/install-tree-sitter.sh v0.22.5
+      - name: Install nodejs
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install Crystal
+        uses: crystal-lang/install-crystal@v1
+        with:
+        crystal: latest
```
Comment (keidax) [inline .github/workflows/ci.yml]:
```suggestion
          crystal: latest
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,19 +2,32 @@ name: CI
...
+
+      - name: Install Tree-sitter
+        run: npm install
```
Comment (keidax) [inline .github/workflows/ci.yml:27]: I admit I don't quite understand what the npm `tree-sitter` package does. How does this install tree-sitter? Does the npm package bundle the main tree-sitter CLI with it?

File: package.json [hunk 3]
```diff
@@ -16,7 +16,7 @@
     "node-gyp-build": "^4.8.0"
   },
   "peerDependencies": {
-    "tree-sitter": "^0.21.1"
+    "tree-sitter": "^0.22.1"
```
Comment (keidax) [inline package.json:19]: Always confuses me that this version doesn't match `tree-sitter --version` :sweat_smile:

File: test/stdlib_coverage [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+stdlib_files.each do |stdlib_file|
+  test = Test.new(stdlib_file, stdlib_file[(stdlib_path.size + 1)..])
+  success = test.execute
+
+  if success
+    pass += 1
+  else
+    failed << test.label
+  end
+  elapsed_ms = "#{test.elapsed.total_milliseconds}ms".colorize.dark_gray
```
Comment (keidax) [inline test/stdlib_coverage]: This lines up the durations more cleanly

```suggestion
  elapsed_ms = sprintf("%8.3fms", test.elapsed.total_milliseconds).colorize.dark_gray
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -2,19 +2,32 @@ name: CI
...
 
       - name: Run tests
-        run: tree-sitter test
+        run: npm test
```
Comment (keidax) [inline .github/workflows/ci.yml]: Why not add `crystal run test/stdlib_coverage` as a new step?

**Review by keidax — COMMENTED** [review]
Hey @hugopl, this looks awesome! I love this idea for testing the parser across a wide range of source files, and the test runner is slick.

File: package.json (hunk 3 above)
Comment (hugopl) [reply package.json:19]: It will only match if you install treesitter with npm.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (hugopl) [reply .github/workflows/ci.yml:27]: It does, from pre-built binaries.

**Review by hugopl — COMMENTED** [review]
Nice, I'll apply the suggestions after the weekend.

**Comment by keidax** [comment]
Heads up, I'm about to merge a big PR to update the tree-sitter version (#18), so you'll need to rebase

File: .github/workflows/ci.yml (hunk 2 above)
Comment (keidax) [reply .github/workflows/ci.yml:27]: Okay, I figured out why I was confused. `tree-sitter` is the Node library for tree-sitter bindings. `tree-sitter-cli` has the CLI binaries.

The current `install-tree-sitter.sh` should function fine here to add `tree-sitter` to the build, so you can remove this step.

File: package.json [hunk 6]
```diff
@@ -5,7 +5,7 @@
   "main": "bindings/node",
   "types": "bindings/node",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "tree-sitter test && crystal run test/stdlib_coverage",
```
Comment (keidax) [inline package.json:8]: The convention for other TS grammars is to use `npm test` for testing the Node language bindings:
https://github.com/tree-sitter/tree-sitter-javascript/blob/108b2d4d17a04356a340aea809e4dd5b801eb40d/package.json#L58

So the coverage test doesn't belong here, but you could add a `test/run_all.sh` script and put it there if you want.

**Comment by hugopl** [comment]
I will close this, so we can everybody join forces on http://github.com/crystal-lang-tools/tree-sitter-crystal

**Pull request closed without merging**